src/include/txn.i

```diff
@@ -897,8 +914,12 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint
   WT_UPDATE *upd, WT_CELL_UNPACK_KV *vpack)
 {
     WT_TIME_WINDOW tw;
+    WT_UPDATE *prepare_upd;
+    uint8_t prepare_state;
+
+    prepare_upd = NULL;
 
-    WT_RET(__wt_txn_read_upd_list(session, cbt, upd));
+    WT_RET(__wt_txn_read_upd_list(session, cbt, upd, &prepare_upd));
```

Comment:
There is a problem in our __wt_txn_read function. consider the below scenario.
1. Prepared update is written to data store.
2. value is read by a transaction, for which WT_TXN_IGNORE_PREPARE is set.
in this case the txn will get the prepared update from the page.

Reply:
It' OK because we always return invisible for onpage prepared value.